### PR TITLE
Fixed #31608 -- Made datetime naive in forms.DateTimeField when USE_TZ is False.

### PR DESCRIPTION
**Pull request by hramezani**
Fix for https://code.djangoproject.com/ticket/31608

### CODE REVIEW

File: tests/forms_tests/field_tests/test_datetimefield.py [hunk 1]
```diff
@@ -41,11 +41,8 @@ def test_datetimefield_clean(self):
             ('2014-09-23T22:34:41', datetime(2014, 9, 23, 22, 34, 41)),
             ('2014-09-23T22:34', datetime(2014, 9, 23, 22, 34)),
             ('2014-09-23', datetime(2014, 9, 23, 0, 0)),
-            ('2014-09-23T22:34Z', datetime(2014, 9, 23, 22, 34, tzinfo=utc)),
-            (
-                '2014-09-23T22:34+07:00',
-                datetime(2014, 9, 23, 22, 34, tzinfo=get_fixed_timezone(420)),
-            ),
+            ('2014-09-23T22:34Z', datetime(2014, 9, 23, 17, 34)),
+            ('2014-09-23T22:34+07:00', datetime(2014, 9, 23, 10, 34)),
```
Comment (hramezani) [inline tests/forms_tests/field_tests/test_datetimefield.py:45]: Not sure about these changes. the ticket is about the datetime object `timezone` and these now don't have timezone

Comment (claudep) [reply tests/forms_tests/field_tests/test_datetimefield.py:45]: We may consider such a change as a data loss. Even if `USE_TZ` is False, I think that a datetime input *with* timezone information should keep the timezone (and hence be aware). I think the docs might need to be updated to say that parsing a datetime with timezone gives an aware datetime even if USE_TZ is False.

File: tests/forms_tests/field_tests/test_datetimefield.py [hunk 2]
```diff
@@ -58,12 +55,33 @@ def test_datetimefield_clean(self):
                 ' 2014-09-23T22:34:41.614804 ',
                 datetime(2014, 9, 23, 22, 34, 41, 614804),
             ),
+            (' 2014-09-23T22:34Z ', datetime(2014, 9, 23, 17, 34)),
+        ]
+        f = DateTimeField()
+        for value, expected_datetime in tests:
+            with self.subTest(value=value):
+                cleaned_value = f.clean(value)
+                self.assertTrue(is_naive(f.clean(cleaned_value)))
```
Comment (smithdc1) [inline tests/forms_tests/field_tests/test_datetimefield.py:64]: Do you need to `clean` the `cleaned_value`? (same applies further down)

File: django/forms/fields.py [hunk 3]
```diff
@@ -473,7 +474,11 @@ def to_python(self, value):
             raise ValidationError(self.error_messages['invalid'], code='invalid')
         if not result:
             result = super().to_python(value)
-        return from_current_timezone(result)
+        if settings.USE_TZ:
+            result = from_current_timezone(result)
+        elif not settings.USE_TZ and timezone.is_aware(result):
```
Comment (smithdc1) [inline django/forms/fields.py:479]: Do we need `not settings.USE_TZ`?

File: tests/forms_tests/field_tests/test_datetimefield.py (hunk 2 above)
Comment (hramezani) [reply tests/forms_tests/field_tests/test_datetimefield.py:64]: No, It remained by mistake.
I am waiting for the decision about this ticket. if the solution got approved I will fix it.
Thanks

File: django/forms/fields.py (hunk 3 above)
Comment (hramezani) [reply django/forms/fields.py:479]: No, we can remove it.

**Comment by hramezani** [comment]
Hi @carltongibson . I will create a PR for doc change soon.

**Comment by carltongibson** [comment]
Hero. Thanks.

**Comment by carltongibson** [comment]
I think some note in 3.1.txt is appropriate.

**Pull request closed without merging**